stdlib/template/autotools.py

```diff
@@ -0,0 +1,138 @@
+        * ``configure``
+        * ``compile``
+        * ``check``
+        * ``install``
+        * ``split``
+        * ``dependency linking``
+
+    For each one of these steps, a function is called. This template simply calls all of them in the above order.
+    All of those functions can be given as arguments, but each one of them have a default value that is explained below.
+    If any of these function is ``None``, the step is skipped.
+
+    **Fetch**
+
+        This step is used to download the source code. The default value is :py:func:`.fetch` with no argument.
+
+    **Extract**
+
+        This step is used to extract the downloaded source code. The default value is :py:func:`.flat_extract_all` with no argument.
+
+    **Patch**
+
+        This step is used to patch the downloaded source code. The default value is :py:func:`.patch_all` with no argument.
+
+    From now one, the current working directory is changed in favor of ``build_folder`` (which defaults to ``.``).
+    If the directory pointed by ``build_folder`` doesn't exist, it is created.
+
+    This is useful for the few ``configure`` scripts that don't work if they aren't executed in a standalone directory.
+
+    **Configure**
+
+        This step uses the ``configure`` script to configure the source code. The default value is :py:func:`.configure` with no argument.
+
+        When ``build_folder`` isn't ``.``, it is usually necessary to override this step with an other call to :py:func:`.configure` with
+        the argument ``binary='../configure'``.
+
+    **Compile**
+
+        This step compiles the source code. The default value is :py:func:`.make` with no argument.
+
+    **Check**
+
+        This step runs the unit and integration tests. The default value is :py:func:`.make` with the arguments ``'check'`` and ``fail_ok=True``.
+
+    **Install**
+
+        This step installs the software in the install cache. The default value is :py:func:`.make` with the arguments ``'install'`` and
+        ``f'DESTDIR={stdlib.build.current_build().install_cache}'``
+
+        If this step is overwritten, the ``DESTDIR`` variable is crucial and should be reused. Otherwise, and unless the ``configure`` script has been
+        configured to work without it, the installation can damage and overwrite parts of the host system.
+
+    **Split**
+
+        This step is still a work in progress. The default value is ``None``.
+
+    **Dependency Linking**
+
+        This step is still a work in progress. The default value is ``None``.
+    """
+    build = stdlib.build.current_build()
+
+    stdlib.log.ilog("Step 1/9: Fetch")
+    if fetch is not None:
+        fetch()
+
+    stdlib.log.ilog("Step 2/9: Extract")
+    if extract is not None:
+        extract()
+
+    stdlib.log.ilog("Step 3/9: Patch")
+    if patch is not None:
+        patch()
+
+    os.makedirs(build_folder, exist_ok=True)
+    with stdlib.pushd(build_folder):
+
+        print(f"OMG I'M IN {os.getcwd()}")
```

Comment:
is this some debug message you forgot to remove? :')

Reply:
Naaaah, what makes you think that? :3